google/gax/api_callable.py

```diff
@@ -41,59 +41,63 @@
 _MILLIS_PER_SECOND = 1000
 
 
-def _add_timeout_arg(a_func, timeout):
-    """Updates a_func so that it gets called with the timeout as its final arg.
+class _RetryableWrapper(object):
+    """A callable to call with either retrying or timeout parameter."""
 
-    This converts a callable, a_func, into another callable with an additional
-    positional arg.
+    def __init__(self, a_func, settings):
+        """Constructor.
 
-    Args:
-      a_func (callable): a callable to be updated
-      timeout (int): to be added to the original callable as it final positional
-        arg.
+        Args:
+          a_func (callable[[req] resp]): an API call.
+          settings (CallSettings): the base settings for the API call.
+        """
+        self._func = a_func
+        self._settings = settings
 
+    def __call__(self, req, options=None, *args, **kwargs):
+        """The API call, customized by the additional CallOptions.
 
-    Returns:
-      callable: the original callable updated to the timeout arg
-    """
+        Args:
+          req (object): the request object.
+          options (gax.CallOptions): the optional value to customize the
+            calling behavior.
+        """
+        settings = self._settings.merge(options)
```

Comment:
`create_api_call` is not the case because, as is discussed in #104, `create_api_call` is called in the constructor like
```
self._create_shelf = gax.create_api_call(stub.create_shelf, defaults['create_shelf'])
```
and then the call options is passed in the individual method invocation, like:
```
def create_shelf(self, options):
   self._create_shelf(req, options, ...)
```

So the result of `create_api_call` should receive a CallOptions instance, and handling the overrides -- that is the role of merge method.